### ISSUE TITLE
Indent - disallow multiline function call with no indent if some of arguments is single line

### ISSUE THREAD

**Issue by Alexsey** [eslint/eslint#10418]
**What rule do you want to change?** indent

**Does this change cause the rule to produce more or fewer warnings?** more

**How will the change be implemented? (New option, new default behavior, etc.)?** default behavior

**Please provide some example code that this change will affect:**

```js
foo([
    1
],
true,
false)
```

**What does the rule currently do for this code?**
Nothing

**What will the rule do after it's changed?**
This would become an error

**Motivation**
1. single-line arguments are not reading just belonged to the function
2. Code like
```js
foo(
true,
false)
```
is illegal now. It should not become legal just because (and only when) another arg is added at position 0

Related to https://github.com/eslint/eslint/issues/10417
